tools/PI/DevHome.PI/ViewModels/WatsonPageViewModel.cs

```diff
@@ -58,6 +52,54 @@ public void UpdateTargetProcess(Process process)
         }
     }
 
+    public void Dispose()
+    {
+        _watsonHelper?.Dispose();
+        GC.SuppressFinalize(this);
+    }
+
+    public void AddNewEntry(DateTime timeGenerated, string moduleName, string executable, string eventGuid)
+    {
+        var newEntry = new WatsonReport(timeGenerated, moduleName, executable, eventGuid);
+        _dispatcher.TryEnqueue(() =>
+        {
+            ReportEntries.Add(newEntry, entry => entry.DateTimeGenerated);
+        });
+    }
+
+    public void UpdateEntry(string eventGuid, string watsonLog, string directoryPath)
+    {
+        _dispatcher.TryEnqueue(() =>
```

Comment:
I'm guessing this is on the dispatcher because RepostedFiles is running on a different threads, ya?